pkg/kratos/service.go

```diff
@@ -325,6 +326,18 @@ func (s *Service) GetSettingsFlow(ctx context.Context, id string, cookies []*htt
 		return nil, nil, err
 	}
 
+	// If a duplicate identifier was detected, kratos returns a 200 response
+	// with a 4000007 error in the rendered ui messages
+	if resp.StatusCode == http.StatusOK {
+		uiMsg := flow.GetUi()
+
+		for _, message := range uiMsg.GetMessages() {
+			if message.GetId() == DuplicateIdentifier {
+				return nil, nil, fmt.Errorf("an account with the same identifier already exists, contact support")
+			}
+		}
+	}
```

Comment:
note for our team in the future: I would love to see these kind of logic wrapped in a function with a proper name that lets everybody know what's needed without the necessity of having comments explaining things. Let's keep this in mind